src/libm/CMakeLists.txt

```diff
@@ -99,48 +141,48 @@ set_target_properties(${TARGET_OBJECTS} ${TARGET_OBJECTS_GNUABI} PROPERTIES
 
 target_compile_definitions(${TARGET_LIBSLEEF}
   PRIVATE DORENAME=1)
-set_target_properties(${TARGET_LIBSLEEF} ${TARGET_LIBSLEEFGNUABI} PROPERTIES
+set_target_properties(${TARGET_LIBSLEEF} PROPERTIES
   VERSION ${SLEEF_VERSION_MAJOR}.${SLEEF_VERSION_MINOR}
   SOVERSION ${SLEEF_SOVERSION}
   C_STANDARD 99)
 
-# Generate the rename headers at build time
-add_custom_command(
-  OUTPUT ${RENAME_HEADERS} ${RENAME_HEADERS_GNUABI}
-  COMMAND ${CMAKE_COMMAND}
-    -DRENAME_HEADERS="${RENAME_HEADERS}"
-    -DRENAME_HEADERS_GNUABI="${RENAME_HEADERS_GNUABI}"
-    -DLOCATIONS_FILE="${PROJECT_BINARY_DIR}/DefineLocations.cmake"
-    -DTARGET_MKRENAME="${TARGET_MKRENAME}"
-    -DTARGET_MKRENAME_GNUABI="${TARGET_MKRENAME_GNUABI}"
-    -P ${SLEEF_SCRIPT_PATH}/GenerateRenameHeader.cmake
-  DEPENDS
-    ${TARGET_MKRENAME}
-    ${TARGET_MKRENAME_GNUABI})
-
-# Generate the sleef header at build time
-set(SLEEF_INCLUDE_HEADER ${CMAKE_BINARY_DIR}/include/sleef.h)
-add_custom_command(
-  OUTPUT ${SLEEF_INCLUDE_HEADER}
-  COMMAND ${CMAKE_COMMAND}
-    -DTARGET_MKRENAME="${TARGET_MKRENAME}"
-    -DSLEEF_ARCH_X86="${SLEEF_ARCH_X86}"
-    -DSLEEF_ARCH_AARCH64="${SLEEF_ARCH_AARCH64}"
-    -DLOCATIONS_FILE="${PROJECT_BINARY_DIR}/DefineLocations.cmake"
-    -P ${SLEEF_SCRIPT_PATH}/GenerateSleefHeader.cmake
-  DEPENDS
-    ${CMAKE_CURRENT_SOURCE_DIR}/sleeflibm.h.org
-    ${TARGET_MKRENAME}
-    ${TARGET_MKDISP})
-
-# Always run the commands to generate the headers
-add_custom_target(headers ALL DEPENDS
-  ${SLEEF_INCLUDE_HEADER}
-  ${RENAME_HEADERS}
-  ${RENAME_HEADERS_GNUABI})
+# --------------------------------------------------------------------
+# Generate include/sleef.h header file
+# --------------------------------------------------------------------
+set(SLEEF_ORG_HEADER ${CMAKE_CURRENT_SOURCE_DIR}/sleeflibm.h.org)
+set(SLEEF_GEN_HEADER ${CMAKE_BINARY_DIR}/include/sleef.h)
+
+add_custom_command(TARGET ${TARGET_HEADERS} 
+  COMMAND cmake -E copy ${SLEEF_ORG_HEADER} ${SLEEF_GEN_HEADER}
+)
+
+foreach(SIMD ${SLEEF_HEADER_LIST})
+  add_custom_command(TARGET ${TARGET_HEADERS} 
+    COMMAND echo Generating sleef.h: ${TARGET_MKRENAME} ${HEADER_PARAMS_${SIMD}}
+    COMMAND ${TARGET_MKRENAME} ${HEADER_PARAMS_${SIMD}} >> ${SLEEF_GEN_HEADER}
+  )
+endforeach()
+
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/sleef_footer.h "#undef IMPORT\n#endif\n")
+if(MSVC)
+  string(REPLACE "/" "\\" sleef_footer_input_file "${CMAKE_CURRENT_BINARY_DIR}/sleef_footer.h")
+  add_custom_command(TARGET ${TARGET_HEADERS} 
+    COMMAND type ${sleef_footer_input_file} >> ${SLEEF_GEN_HEADER}
+  )
+else()
+  add_custom_command(TARGET ${TARGET_HEADERS} 
+    COMMAND cat ${CMAKE_CURRENT_BINARY_DIR}/sleef_footer.h >> ${SLEEF_GEN_HEADER}
+  )
+endif()
+
+
 
 # Install libsleef and sleef.h
 install(FILES ${SLEEF_INCLUDE_HEADER}
   DESTINATION include)
-install(TARGETS ${TARGET_LIBSLEEF} ${TARGET_LIBSLEEFGNUABI}
+if(SLEEF_ENABLE_GNUABI)
```

Comment:
We want to install `libsleef` unconditionally. Please change this to:

```
if(GNUABI)
  install(libsleefgnuabi)
endif()

install (libsleef)
```

Reply:
Fixed by push+squash on 0de75b5